lib/fluent/plugin/out_kafka_buffered.rb

```diff
@@ -167,28 +185,29 @@ def write(chunk)
 
         record_buf = @formatter_proc.call(tag, time, record)
         record_buf_bytes = record_buf.bytesize
-        if messages.length > 0 and messages_bytes + record_buf_bytes > @kafka_agg_max_bytes
-          log.on_trace { log.trace("#{messages.length} messages send.") }
-          @producer.send_messages(messages)
-          messages = []
+        if messages > 0 and messages_bytes + record_buf_bytes > @kafka_agg_max_bytes
```

Comment:
You are right. Another way is `(messages > 0) and (messages_bytes + record_buf_bytes > @kafka_agg_max_bytes)`